src/ingestor.ts

```diff
@@ -141,7 +141,14 @@ export default async function ingestor(body) {
   let pinned;
   let receipt;
   try {
-    [pinned, receipt] = await Promise.all([pin(body), issueReceipt(body.sig)]);
+    const { addrress, sig, ...restBody } = body;
+    const ipfsBody = {
+      addrress,
+      sig,
+      hash: id,
+      ...restBody
+    };
+    [pinned, receipt] = await Promise.all([pin(ipfsBody), issueReceipt(body.sig)]);
```

Comment:
```suggestion
    const { sig } = body;
    const ipfsBody = {
      hash: id,
      ...restBody
    };
    [pinned, receipt] = await Promise.all([pin(ipfsBody), issueReceipt(body.sig)]);
```

Also, I don't think we can verify this signature without removing the `hash` from params, as mentioned in 
https://github.com/snapshot-labs/snapshot-sequencer/pull/47#issuecomment-1544561913 you tried?